lib/ansible/module_utils/facts/system/ssh_pub_keys.py

```diff
@@ -47,6 +47,9 @@ def collect(self, module=None, collected_facts=None):
                 key_filename = '%s/ssh_host_%s_key.pub' % (keydir, type_)
                 keydata = get_file_content(key_filename)
                 if keydata is not None:
-                    ssh_pub_key_facts[factname] = keydata.split()[1]
+                    (key_type, keydata) = keydata.split(None, 1)
+                    ssh_pub_key_facts[factname] = keydata
+                    type_factname = '{}_type'.format(factname)
```

Comment:
This doesn't create the fact, at least not on my machine. I created a PR with my proposed (and tested) fix and referenced the issue. Mine could be expanded for all key types like it looks like you're shooting for here; however, if ECDSA is the only type with sub-types, then I'd think that's all we need.